src/lib/fragments/AgGrid.react.js

```diff
@@ -555,7 +554,7 @@ export default class DashAgGrid extends Component {
         const {setProps, dangerously_allow_code} = this.props;
 
         const setCellProps = (props) => {
-            setProps({clickData: props.clickData, hoverData: props.hoverData});
+            setProps({cellRendererData: props.cellRendererData});
```

Comment:
💄 
```js
const setCellProps = ({cellRendererData}) => {
    setProps({cellRendererData});
}
```

Makes me wonder though - now that it's clear that this is only going to a single prop, should we pass it to renderers as a `setData` prop rather than `setProps`, ie in [rowMenuRenderer](https://github.com/plotly/dash-ag-grid/pull/70/files#diff-4aa177d562cc6efb214ba041c20342ed5bb67d7a994c0d7f8d8d00a5365e612fR25) instead of `setProps({cellRendererData: {value, data}})` you would just call `setData({value, data})`, then the function here would just be:

```js
const setCellProps = (cellRendererData) => {
    setProps({cellRendererData});
}
```

But then I also wonder if this is enough for real use cases... would we want to add info about which cell this is, as well as `timestamp` like we have on `clickData`?

Reply:
As far as that info, it’s up to the dev to decide what they want to use or provide. 

Unless we automatically tack on a timestamp? Like cellRendererData_timestamp: Date.now()?

Reply:
See: https://github.com/plotly/dash-ag-grid/pull/70/commits/6f07a05ae567dce907f4633ea8cba903d8959d1f

Reply:
Sorry, I misspoke, I meant to refer to `onCellClicked`, where we call:
```js
setProps({cellClicked: {value, colId, rowIndex, timestamp}})
```
`colId` and `rowIndex` give you an exact pointer to the cell that was clicked (would be even more useful if we add `rowId` to this, at least when `getRowId` is defined), `timestamp` isn't generally intended to really be used, just ensures uniqueness, and `value` is the cell data you clicked on.

So I feel like we should follow that same pattern here. When defining a custom renderer, you just call `setData(arbitraryObject)` but you shouldn't need to explicitly include the extra info needed to identify the cell and make the object unique. we should take your `arbitraryObject` and set it as the `value` inside `cellRendererData`, but also include `colId, rowIndex, rowId, timestamp`.

Look at the row menu renderer example https://dashaggrid.pythonanywhere.com/components/row-menu: The renderer chooses to include the full row data, which if they're unique allows you to identify which row you're in. If they aren't unique you can't identify the row, and it does nothing to tell you which column the event came from so if you had these menus in multiple columns with matching option labels you'd need to build uniqueness into the `value` field. If the table has lots of columns, providing the full row is also more info than you generally need. I'd love if `rowMenuRenderer` could just call `setData(value)` and all the other pieces (`colId, rowIndex, rowId, timestamp`) are added automatically. That would make any additional renderers people make more powerful without needing to think about any of this.

AFAICT if we move `setCellProp` into the function `return (props) => ( <Renderer ... /> )` we have all the info we need to do this? And then of course with `timestamp` built into the `cellRendererData` prop we don't need the `cellRendererData_timestamp` prop anymore.

Reply:
see https://github.com/plotly/dash-ag-grid/pull/70/commits/f662c2ee71142a5fb901eb570f7d65522e5502e6